crates/fuzzing/src/generators/table_ops.rs

```diff
@@ -85,29 +117,22 @@ pub(crate) enum TableOp {
 }
 
 impl TableOp {
-    fn to_wat_string(&self, wat: &mut String) {
+    fn insert(&self, func: &mut Function) {
         match self {
             Self::Gc => {
-                wat.push_str("(call 0)");
+                func.instruction(Instruction::Call(0));
             }
             Self::Get(x) => {
-                wat.push_str("(drop (table.get $table (i32.const ");
-                write!(wat, "{}", x).unwrap();
-                wat.push_str(")))");
+                func.instruction(Instruction::Drop);
+                func.instruction(Instruction::TableGet { table: *x });
             }
             Self::SetFromParam(x, y) => {
-                wat.push_str("(table.set $table (i32.const ");
-                write!(wat, "{}", x).unwrap();
-                wat.push_str(") (local.get ");
-                write!(wat, "{}", y).unwrap();
-                wat.push_str("))");
+                func.instruction(Instruction::TableSet { table: *x });
+                func.instruction(Instruction::LocalGet((*y).into()));
             }
             Self::SetFromGet(x, y) => {
-                wat.push_str("(table.set $table (i32.const ");
-                write!(wat, "{}", x).unwrap();
-                wat.push_str(") (table.get $table (i32.const ");
-                write!(wat, "{}", y).unwrap();
-                wat.push_str(")))");
+                func.instruction(Instruction::TableSet { table: *x });
+                func.instruction(Instruction::TableGet { table: *y });
```

Comment:
Similarly, these need to be swapped around as well so that the `table.get` comes before the `{local,table}.set`.

Reply:
Also, the `table.{get,set}` should take the table index as a Wasm stack operand (created via `i32.const`) not as an immediate that is part of the instruction (which specifies which table is being used):

```
// Stack: []
func.instruction(Instruction::I32Const(*y);
// Stack: [y]
func.instruction(Instruction::I32Const(*x));
// Stack: [y, x]
func.instruction(Instruction::TableGet { table: 0 });
// Stack: [y, ref0]
func.instruction(Instruction::TableSet { table: 0 });
// Stack: []
```

Reply:
Thanks for clarifying this, I was confused what `table: ` was referring to (table index or value to put/get into a table).